lib/ddtrace/profiling/events/stack.rb

```diff
@@ -0,0 +1,74 @@
+require 'ddtrace/profiling/event'
+
+module Datadog
+  module Profiling
+    module Events
+      # Describes a stack profiling event
+      class Stack < Event
+        attr_reader \
+          :frames,
+          :total_frame_count,
+          :thread_id
+
+        def initialize(
```

Comment:
Decided to force these arguments to be specified like a Struct, rather than permitting Hash options for a couple of reasons:
 - These attributes should be available at initialization time and are "required"
 - Its less performant to create many Hashes to ferry options into the object. Considering this will be rapidly generated, I wanted to make sure its as lean as reasonably possible.